ouroboros-consensus/test-consensus/Test/Dynamic/General.hs

```diff
@@ -47,7 +47,6 @@ runTestNetwork ::
   forall blk.
      ( RunNode blk
      , TxGen blk
-     , TracingConstraints blk
```

Comment:
Hmm, this one (and the one in `Network.hs`) is there to make it easy to replace a `nullTracer` with a `showTracing debugTracer` for debugging. So while it is indeed redundant, it is intentional.

Reply:
Ok, addressed this in a hopefully more principled way.